scenes/backend/backend.gd

```diff
@@ -21,6 +21,30 @@ var server_key = "projectlam"
 @onready var nLoginFailedDialog: Control = find_child("LoginFailedDialog")
 @onready var nLoginFailedDialog_AnimationPlayer: AnimationPlayer = nLoginFailedDialog.get_node("AnimationPlayer")
 
+signal full_initialization
+signal received_match_presence(p_presence: NakamaRTAPI.MatchPresenceEvent)
+signal received_match_state(p_state)
+signal peer_connected(id : int)
+
+# use 'await await_finit()' to wait for full initialization.
+func await_finit():
+	if _is_fully_initialized:
+		return
+	await full_initialization
+
+func await_connection():
+	await await_finit()
+	while(
+		(socket.is_connected_to_host() || socket.is_connecting_to_host()) &&
+		multiplayer.multiplayer_peer.get_connection_status() == MultiplayerPeer.CONNECTION_CONNECTING):
+		print_debug(socket._adapter._ws.get_ready_state())
+		await get_tree().process_frame
```

Comment:
Is Godot not errorring on this? I thought you'd get an indentation error here, isn't that based on the tab level of the line that has the colon? (was that a change in Godot 4?)

Reply:
I don't think so. When you use parentheses you can ignore all indentations, newlines and stuff till you close the parentheses.
It's not as great as other languages, but it let's you break down a statement to multiple lines. I think i should adding an empty line to improve the readability though.
Think of it as how you can break down a long function call to multiple lines.

Of course i should've removed that line 40's debug message which is useless at this point. Will do that in next commits. Right now i want this pull request to go through. the later it goes through, the more conflicts will be.

Reply:
Yeah looks like you're right. I don't know why I thought it depended on the colon, can't get it to error in either Godot 3 or 4.